build_product_images.py

```diff
@@ -125,48 +127,97 @@ def build_image_tags(image_name: str, image_version: str, product_version: str)
     Returns the --tag command line arguments that are used by the docker build command.
     """
     return [
-        "-t",
         f"{image_name}:{product_version}-stackable{image_version}",
     ]
 
 
-def build_and_publish_image(args: Namespace, product_name: str, versions: Dict[str, str]) -> List[List[str]]:
+def generate_bakefile(args: Namespace):
     """
-    Returns a list of commands that need to be run in order to build and
-    publish product images.
+    Generates a Bakefile (https://docs.docker.com/build/bake/file-definition/) describing how to build the whole image graph.
 
-    For local building, builder instances are supported.
+    build_and_publish_images() ensures that only the desired images are actually built.
+    """
+    targets = {}
+    groups = {}
+    contexts = {}
+    products = {product["name"]: product for product in conf.products}
+    product_names = list(products.keys())
+    for product_name, product in products.items():
```

Comment:
```suggestion
    product_names = [p.get("name") for p in conf.products]
    for product_name, product in zip(product_names, conf.products):
```

Reply:
This is an old leftover from the context mechanism, I'll go simplify.